### DISCUSSION TITLE
NestJS Singular Mapper

### DISCUSSION THREAD

**Discussion by BeyondBelief96** [nartc/mapper#425] [Q&A] [answered]
Hello, I had a quick question about how to use a singular mapper in NestJS. I was looking at the docs and couldn't find the answer. 

I have my AppModule registering a single mapper like so: 

![image](https://user-images.githubusercontent.com/50305220/156035435-0dbd981a-c70f-4962-b4fa-0ea126e2f5e8.png)

and I have my mapping profile like so: 

![image](https://user-images.githubusercontent.com/50305220/156035556-193953bd-855d-4d4c-88af-e54403abc020.png)

I want to use the mapper in the OpportunityModule. I have injected the mapper into my OpportunityService like so: 
![image](https://user-images.githubusercontent.com/50305220/156035669-574bd236-dc46-4312-9a4b-c6b839c9c237.png)

But I get dependency errors saying that it can't resolve the mapper. My OpportunityModule looks like this: 
![image](https://user-images.githubusercontent.com/50305220/156035745-f197493d-1f24-437c-8ace-acac3b1b2348.png)

What am I missing here? How do I get the mapper that's initialized at the AppModule level to resolve at the sub module level? For reference, here is the specific error I am recieving:

[Nest] 81886  - 02/28/2022, 12:06:57 PM   ERROR [ExceptionHandler] Nest can't resolve dependencies of the OpportunityService (OpportunityRepository, ?). Please make sure that the argument automapper:nestjs:dto-mapper at index [1] is available in the OpportunityModule context.

Potential solutions:
- If automapper:nestjs:dto-mapper is a provider, is it part of the current OpportunityModule?
- If automapper:nestjs:dto-mapper is exported from a separate @Module, is that module imported within OpportunityModule?
  @Module({
    imports: [ /* the Module containing automapper:nestjs:dto-mapper */ ]
  })

**Comment by nartc** [comment, marked as answer]
Use `@InjectMapper()` without the argument.

**Reply by BeyondBelief96** [reply]
Wow, thanks for that. I saw that it was "optional" in the docs and assumed that wouldn't cause an issue.

**Reply by nartc** [reply]
Let me clarify. The context here is that you can (technically) have many different Mappers in one application. So when you provide the options with `AutomapperModule.forRoot()`, you provide it as an `Array`. And the `name` is the identifier of the Mapper. So you might have:

```ts
@InjectMapper('classes-mapper') private classMapper: Mapper,
@InjectMapper('pojos-mapper') private pojosMapper: Mapper
```

More than often though, you only have one Mapper in one application. `singular:true` will change the behavior of `@InjectMapper()`
